engineio/transport/polling/bufreader.go

```diff
@@ -0,0 +1,75 @@
+package polling
+
+import (
+	"io"
+)
+
+// bufReader is similar to bufio.Reader, with a customed buffer.
+type bufReader struct {
+	buf        []byte
+	start, end int
+	rd         io.Reader
+	finished   bool
+}
+
+func newBufReader(buf []byte, rd io.Reader) *bufReader {
```

Comment:
why are you need set buffer into constructor?

Reply:
The buffer is generated by the caller. It could let the caller reuse a buffer when a transport closed.

Reply:
hm... why not?!